dwave/cloud/coders.py

```diff
@@ -278,52 +283,52 @@ def decode_qp_numpy(msg, return_matrix=True):
     return result
 
 
-def _encode_problem_as_bq_ref(problem):
-    assert isinstance(problem, str)
-
-    return problem
+def encode_problem_as_ref(problem):
+    """Encode the problem given via reference for submission in the `ref` data
+    format.
 
-def _encode_problem_as_bq_json(problem):
-    assert hasattr(problem, 'to_serializable')
+    Args:
+        problem (str):
+            A reference to an uploaded problem (Problem ID).
 
-    return problem.to_serializable(use_bytes=False)
+    Returns:
+        encoded submission dictionary
+    """
 
-def _encode_problem_as_bq_json_zlib(problem):
-    assert hasattr(problem, 'to_serializable')
+    if not isinstance(problem, str):
+        raise TypeError("unsupported problem type")
 
-    bqm_dict = _encode_problem_as_bq_json(problem)
-    return zlib.compress(codecs.encode(json.dumps(bqm_dict), "ascii"))
+    return {
+        'format': 'ref',
+        'data': problem
+    }
 
 
-def encode_problem_as_bq(problem, compress=False):
+def encode_problem_as_bq(problem):
     """Encode the binary quadratic problem for submission in the `bq` data
     format.
 
     Args:
-        problem (:class:`~dimod.BinaryQuadraticModel`/str):
-            A binary quadratic model, or a reference to one (Problem ID).
+        problem (:class:`~dimod.BinaryQuadraticModel`):
+            A binary quadratic model.
 
     Returns:
         encoded submission dictionary
-    """
 
-    if isinstance(problem, str):
-        return {
-            'format': 'bqm-ref',
-            'data': _encode_problem_as_bq_ref(problem)
-        }
+    Note:
+        The `bq` format assumes the complete BQM is sent embedded in the sample
+        job submit data, something none of the production solvers currently
```

Comment:
"the sample job submit data" unclear (to me at least) what this refers to

Reply:
Refers to problem (BQM) data encoding when submitting a (sampling) job to SAPI.